File: src/app/models/reply/ReplyResponse.ts
  export interface ReplyResponse {
    id: number;
    discussionId: number;
    //forumId: number;
    messageContent: string;
    authorId: number;
    authorFullName: string;
    parentAuthorId: number;
    createdAt: string;
    updatedAt: string;
    replies?: ReplyResponse[]; // Nested replies
    parentAnswerId?: number;
    showReplyInput?: boolean; // Define showReplyInput property

  }

File: src/main/java/com/pfe/elearning/forum/answer/dto/ReplyResponse.java
package com.pfe.elearning.forum.answer.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReplyResponse {
    private Integer id;
    private String messageContent;
    private Integer authorId;
    private String authorFullName;
    private Integer parentAnswerId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<ReplyResponse> replies;

}
